step-framework-parent/step-framework-collections/src/main/java/step/core/accessors/Accessor.java

```diff
@@ -83,10 +83,18 @@ public interface Accessor<T extends AbstractIdentifiableObject> {
 	Iterator<T> getAll();
 	
 	/**
+	 * Warning: depending on the underlying implementation the whole result set might be loaded into memory
+	 * Use streamCloseable instead within a try-with-resources statement for large volume of data
 	 * @return a {@link Stream} with all objects
 	 */
 	Stream<T> stream();
 
+	/**
+	 * Warning: this method must be used within a try-with-resources statement or similar control structure to ensure that the stream's I/O resources are closed promptly after the stream's operations have completed.
+	 * @return a {@link Stream} with all objects
+	 */
+	Stream<T> streamCloseable();

```

Comment:
Also not sure about the naming but I don't have a better idea. The main difference with the legacy method is that it is performing "real" streaming. Don't know how to reflect this in the name

Reply:
renamed to fetchStream

Reply:
renamed to fetchLazy